### PR TITLE
Create the ECS release checklist

### PR DESCRIPTION
**Pull request by webmat**

### CODE REVIEW

File: meta/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Release Checklist
```
Comment (ruflin) [inline meta/release_checklist.md]: Should we put this also in `docs` for now? I hesitant to introduce a new directory.

Comment (webmat) [reply meta/release_checklist.md]: I viewed `docs/` as the pieces of the readme, not a general place to put documentation. But I don't mind moving it there :-)

Comment (ruflin) [reply meta/release_checklist.md]: Let's move it there for now until we figured out a better place. I'm mainly worried about the naming `meta` :-)

**Comment by webmat** [comment]
Closing as we've come up with a more complete release checklist elsewhere.

**Pull request closed without merging**